apps/server/src/modules/task/uc/task.uc.spec.ts

```diff
@@ -91,449 +70,401 @@ describe('TaskUC', () => {
+					return { user, task, spy };
+				};
 
-				await service.findAllFinished(user.id);
+				it('should select the right status', async () => {
+					const { user, spy } = setup();
 
-				expect(spy).toHaveBeenCalled();
+					await service.findAllFinished(user.id);
 
-				mockRestore();
+					expect(spy).toHaveBeenCalled();
+					spy.mockRestore();
+				});
 			});
 		});
 
 		describe('when user has no task overview permissions', () => {
-			beforeEach(() => {
+			const setup = () => {
 				const permissions = [];
-				user = setupUser(permissions);
-			});
+				const user = setupUser(permissions);
 
-			it('should fail with UnauthorizedException', async () => {
-				const mockRestore = findAllMock({});
-				authorizationService.checkOneOfPermissions.mockImplementation(() => {
+				authorizationService.checkOneOfPermissions.mockImplementationOnce(() => {
 					throw new UnauthorizedException();
 				});
-				await expect(() => service.findAllFinished(user.id)).rejects.toThrow(UnauthorizedException);
 
-				mockRestore();
+				return { user };
+			};
+
+			it('should fail with UnauthorizedException', async () => {
+				const { user } = setup();
+
+				await expect(() => service.findAllFinished(user.id)).rejects.toThrow(UnauthorizedException);
 			});
 		});
 	});
 
 	describe('findAll', () => {
-		beforeEach(() => {
-			const permissions = [Permission.TASK_DASHBOARD_VIEW_V3];
-			user = setupUser(permissions);
-		});
+		describe('without permissions', () => {
+			const setup = () => {
+				const permissions = [];
+				const user = setupUser(permissions);
+				const paginationParams = new PaginationParams();
 
-		it('should call authorizationService.hasAllPermissions with [Permission.TASK_DASHBOARD_VIEW_V3]', async () => {
-			const permissions = [];
-			user = setupUser(permissions);
-			const mockRestore = findAllMock();
+				authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+				authorizationService.hasAllPermissions.mockReturnValueOnce(false);
 
-			const paginationParams = new PaginationParams();
-			await service.findAll(user.id, paginationParams);
-			expect(authorizationService.hasAllPermissions).toBeCalledWith(user, [Permission.TASK_DASHBOARD_VIEW_V3]);
-			mockRestore();
-		});
+				return { user, paginationParams };
+			};
 
-		it('should call authorizationService.hasAllPermissions with [Permission.TASK_DASHBOARD_TEACHER_VIEW_V3]', async () => {
-			const permissions = [];
-			user = setupUser(permissions);
-			const mockRestore = findAllMock();
-			authorizationService.hasAllPermissions.mockReturnValueOnce(false);
-			const paginationParams = new PaginationParams();
-			await service.findAll(user.id, paginationParams);
-			expect(authorizationService.hasAllPermissions).toBeCalledWith(user, [Permission.TASK_DASHBOARD_TEACHER_VIEW_V3]);
-			mockRestore();
-		});
+			it('should call authorizationService.hasAllPermissions with correct permissions and throw', async () => {
```

Comment:
I think if you test for the correct exception you don't need to test the calls to the auth service.

Reply:
Not in focus here. Moved to miro board as todo